### PR TITLE
Add the ability to cross-compile using the makefile

### PR DESCRIPTION
**Pull request by aspenluxxxy**
`make TARGET=aarch64-apple-ios`, for example.

### CODE REVIEW

**Comment by sylvestre** [comment]
Could you please add a test for this?
a new check in https://github.com/uutils/coreutils/blob/master/.github/workflows/CICD.yml should be easy :)
thanks

**Comment by aspenluxxxy** [comment]
@sylvestre As a part of an existing workflow, or a small new test entirely?

**Comment by sylvestre** [comment]
I feel that cross compilation, as a new thing, should have a separate check, wdyt?

**Comment by aspenluxxxy** [comment]
True. I'll add it later today; what would be a good target to test cross-compiling to, or should it just test that the `TARGET` variable works?

**Comment by sylvestre** [comment]
maybe x86 ? 
and some less common arch like s390x?

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -567,3 +567,31 @@ jobs:
...
+  cross-compile:
+    name: Cross-Compile
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        job:
+          - { toolchain: x86_64-unknown-linux-gnu, target: x86_64-unknown-linux-gnu }
+          - { toolchain: x86_64-unknown-linux-gnu, target: aarch64-unknown-linux-gnu }
+          - { toolchain: x86_64-unknown-linux-gnu, target: arm-unknown-linux-gnueabihf }
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:579]: could you please also add s390x? thanks

Comment (aspenluxxxy) [reply .github/workflows/CICD.yml:579]: what's the ubuntu packages for the s390x toolchain

Comment (aspenluxxxy) [reply .github/workflows/CICD.yml]: nvm

**Comment by aspenluxxxy** [comment]
well... the tests fail, but the makefile works.

**Comment by sylvestre** [comment]
It is expected as you need the hardware (or emulator to run it).
Having just the build would be enough.

**Comment by aspenluxxxy** [comment]
i mean the cross-compilation test fails.

**Comment by sylvestre** [comment]
@aspenluxxxy are you still planning to work on this? thanks

**Comment by aspenluxxxy** [comment]
wlel id on't know howe to fix it

**Comment by sylvestre** [comment]
Well, we could try to make it work :)

**Pull request closed without merging**